x-pack/plugins/maps/public/classes/layers/tiled_vector_layer/tiled_vector_layer.tsx

```diff
@@ -188,6 +262,30 @@ export class TiledVectorLayer extends VectorLayer {
     this._setMbCentroidProperties(mbMap, sourceMeta.layerName);
   }
 
+  queryForTileMeta(mbMap: MbMap): Feature[] | null {
+    // @ts-ignore
+    const mbSource = mbMap.getSource(this._getMbSourceId());
+    if (!mbSource) {
+      return null;
+    }
+
+    const sourceDataRequest = this.getSourceDataRequest();
+    if (!sourceDataRequest) {
+      return null;
+    }
+    const sourceMeta: MVTSingleLayerVectorSourceConfig = sourceDataRequest.getData() as MVTSingleLayerVectorSourceConfig;
+    if (sourceMeta.layerName === '') {
+      return null;
+    }
+
+    const mbFeatures = mbMap.querySourceFeatures(this._getMbSourceId(), {
+      sourceLayer: sourceMeta.layerName,
+      filter: ['==', ['get', KBN_METADATA_FEATURE], true],
+    });
+
+    return mbFeatures as Feature[];
```

Comment:
querySourceFeatures API docs say "Because features come from tiled vector data or GeoJSON data that is converted to tiles internally, feature geometries may be split or duplicated across tile boundaries and, as a result, features may appear multiple times in query results." Can you add a comment explaining that since these are meta per tile the features will never cross tile boundaries. Something like "querySourceFeatures can return duplicated features when features cross tile boundaries. Tile meta will never have duplicated features since by there nature, tile meta is a feature contained within a single tile"

Reply:
discussed offline. This is not the case here, since metadata-features do not cross tile-boundaries. So there's no possibility for duplication in this particular instance.

Reply:
But can you at least add a comment saying this.

"querySourceFeatures can return duplicated features when features cross tile boundaries. Tile meta will never have duplicated features since by there nature, tile meta is a feature contained within a single tile"